extensions/platform-http/runtime/src/main/java/org/apache/camel/quarkus/component/platform/http/runtime/QuarkusPlatformHttpConsumer.java

```diff
@@ -0,0 +1,376 @@
+                    writeResponse(ctx, e, headerFilterStrategy);
+                } catch (Exception e1) {
+                    LOG.debugf(e1, "Could not handle '%s'", path);
+                    ctx.fail(e1);
+                }
+            });
+
+        this.route = r;
+    }
+
+    @Override
+    protected void doStop() throws Exception {
+        if (route != null) {
+            route.remove();
+            route = null;
+        }
+        super.doStop();
+    }
+
+    @Override
+    protected void doSuspend() throws Exception {
+        if (route != null) {
+            route.disable();
+        }
+        super.doSuspend();
+    }
+
+    @Override
+    protected void doResume() throws Exception {
+        if (route != null) {
+            route.enable();
+        }
+        super.doResume();
+    }
+
+    Object toHttpResponse(HttpServerResponse response, Message message, HeaderFilterStrategy headerFilterStrategy) {
+        final Exchange exchange = message.getExchange();
+        final boolean failed = exchange.isFailed();
+        final int defaultCode = failed ? 500 : 200;
+
+        final int code = message.getHeader(Exchange.HTTP_RESPONSE_CODE, defaultCode, int.class);
+
+        response.setStatusCode(code);
+
+        final TypeConverter tc = exchange.getContext().getTypeConverter();
+
+        //copy headers from Message to Response
+        if (headerFilterStrategy != null) {
+            for (Map.Entry<String, Object> entry : message.getHeaders().entrySet()) {
+                final String key = entry.getKey();
+                final Object value = entry.getValue();
+                // use an iterator as there can be multiple values. (must not use a delimiter)
+                final Iterator<?> it = ObjectHelper.createIterator(value, null);
+                String firstValue = null;
+                List<String> values = null;
+                while (it.hasNext()) {
+                    final String headerValue = tc.convertTo(String.class, it.next());
+                    if (headerValue != null
+                            && !headerFilterStrategy.applyFilterToCamelHeaders(key, headerValue, exchange)) {
+                        if (firstValue == null) {
+                            firstValue = headerValue;
+                        } else {
+                            if (values == null) {
+                                values = new ArrayList<String>();
+                                values.add(firstValue);
+                            }
+                            values.add(headerValue);
+                        }
+                    }
+                }
+                if (values != null) {
+                    response.putHeader(key, values);
+                } else if (firstValue != null) {
+                    response.putHeader(key, firstValue);
+                }
+            }
+        }
+
+        Object body = message.getBody();
+        final Exception exception = exchange.getException();
```

Comment:
I think in some of the other http components we have options to turn this on|off as we should not leak server side exceptions by default to clients

Reply:
Would you plz. file a followup issue?

Reply:
Ah netty-http component dont have this feature

Reply:
Ah it was the transfer exception option. I created a new ticket for not dumping stacktraces so end users can turn this off for production usage etc: https://issues.apache.org/jira/browse/CAMEL-14025

Reply:
So if there anything to do here, please file a followup issue.